TheMET/TheMET/Network/ImageLoader.swift

```diff
@@ -10,22 +10,30 @@ import UIKit
 
 class ImageLoader {
     
+    private var loadedImagesCollection: [URL : UIImage] = [:]
+    
     func loadImage(urlString: String, completion: @escaping (UIImage?) -> Void) {
+        
         guard let imageURL = URL(string: urlString) else {
             completion(nil)
             return
         }
+        if let image = self.loadedImagesCollection[imageURL]  {
+            completion(self.loadedImagesCollection[imageURL])
```

Comment:
Just use the already extracted `image` instead of `self.loadedImagesCollection[imageURL]`